### ISSUE TITLE
Optimization the output for the suggested mirrors

### ISSUE THREAD

**Issue by madanyang** [bmwiedemann/bench-http#2]
Rather than continuing on issue #1 I would like to propose another optimization:

- run benchlist test a couple of reasonable times i.e 3 or let the user define maybe with an upper limit 
- combine the test results 
- take the averages of RTT and Dataspeed
- Either provide both or let the user choose what is important for them.

How does that sound ?

**Comment by bmwiedemann** [comment]
Question is how much that will improve data quality. One thing that can distort measurement is initial DNS lookup. In zypper that will only happen once and then will be cached in a local resolver.
To counter that, it would be enough to only measure RTT twice and take the 2nd or lower value.

There is already a [`metric` function](https://github.com/bmwiedemann/bench-http/blob/master/findbestmirror.pl#L26) defined  that combines RTT and dataspeed. We could add documentation or some easier way to tweak that.

**Comment by madanyang** [comment]
You are right, I have seen the function afterwards, why not take average to be on the safe side.

As a side note [bench-one.pl](https://github.com/bmwiedemann/bench-http/blob/master/benchlist.pl#L19) is been hard coded, why not make it portable so if the whole thing is in users bin/ directory or in /usr/local/bin it would work with out a hickup.

But hey thanks for the initiative, it sure helps to speed up the install / update chain

**Comment by bmwiedemann** [comment]
commit b2e39bd20a83a7f726643c3094e6f3eec3c5b03d added the DNS warmup. In general, I want to keep the complexity low to ensure correctness.

I'm still not happy with the UX, though.

**Comment by bmwiedemann** [comment]
> As a side note bench-one.pl is been hard coded, why not make it portable so if the whole thing is in users bin/ directory or in /usr/local/bin it would work with out a hickup.

solved with commit babd29dd1faa795aa91dc1949992c67b2b3354cf

**Comment by bmwiedemann** [comment]
> You are right, I have seen the function afterwards, why not take average to be on the safe side.

The `metric` function is already a weighted average to put more emphasis on RTT. This is so because the test uses a 10MB get request but most packages are smaller than that (even smaller than 1MB), so RTT will have a larger impact.

Or did you mean average over multiple measurements? Then the measurement would take longer and the server can cheat because it can serve from RAM and maybe that is not what we want to measure and optimize for.
